pkg/apis/pravega/v1alpha1/pravegacluster_types.go

```diff
@@ -106,22 +106,26 @@ func (s *ClusterSpec) withDefaults() (changed bool) {
 
 	if s.Version == "" {
 		changed = true
-		s.Version = DefaultPravegaVersion
+		if s.Pravega != nil && s.Pravega.Image != nil && s.Pravega.Image.Tag != "" {
```

Comment:
Hmm, so if users don't specify `version`, instead they specify `pravega tag` and `bookkeeper tag`, the `pravega tag` will override the `bookkeeper tag`, which might make users confused. Can we deprecate `tag` in the future? We are still in alpha version, so in theory, we can break any API we want. What do you think?

Reply:
Right. One of the ideas behind this, apart from maintaining backwards compatibility, is that Pravega and BookKeeper images are both based on the same version to prevent users users from mistakenly using different versions of Pravega and BookKeeper, resulting in a potentially incompatible cluster. 

I added a deprecation note to the `Tag` definition in `pravegacluster_types.go`, and I'm also going to add it to the example CR.